File: riveredge-stem/src/services/savedSearch.ts
/**
 * 保存搜索条件服务
 * 
 * 提供保存搜索条件的 API 调用
 */

import { apiRequest } from './api';

/**
 * 保存搜索条件接口
 */
export interface SavedSearch {
  id: number; // 自增ID（内部使用，性能优先）
  uuid: string; // 业务ID（UUID，对外暴露，安全且唯一）
  tenant_id?: number;
  user_id: number;
  page_path: string;
  name: string;
  is_shared: boolean;
  is_pinned: boolean;
  search_params: Record<string, any>;
  created_at: string;
  updated_at: string;
}

/**
 * 创建保存搜索条件请求参数
 */
export interface CreateSavedSearchParams {
  page_path: string;
  name: string;
  is_shared: boolean;
  is_pinned?: boolean;
  search_params: Record<string, any>;
}

/**
 * 更新保存搜索条件请求参数
 */
export interface UpdateSavedSearchParams {
  name?: string;
  is_shared?: boolean;
  is_pinned?: boolean;
  search_params?: Record<string, any>;
}

/**
 * 保存搜索条件列表响应
 */
export interface SavedSearchListResponse {
  total: number;
  items: SavedSearch[];
}

/**
 * 获取保存的搜索条件列表
 * 
 * @param pagePath - 页面路径
 * @param includeShared - 是否包含共享搜索条件（默认 true）
 * @returns 搜索条件列表
 */
export async function getSavedSearchList(
  pagePath: string,
  includeShared: boolean = true
): Promise<SavedSearchListResponse> {
  const response = await apiRequest<SavedSearchListResponse>(
    '/saved-searches',
    {
      method: 'GET',
      params: {
        page_path: pagePath,
        include_shared: includeShared,
      },
    }
  );
  return response;
}

/**
 * 创建保存搜索条件
 * 
 * @param data - 创建参数
 * @returns 创建的搜索条件
 */
export async function createSavedSearch(
  data: CreateSavedSearchParams
): Promise<SavedSearch> {
  const response = await apiRequest<SavedSearch>(
    '/saved-searches',
    {
      method: 'POST',
      data,
    }
  );
  return response;
}

/**
 * 获取单个保存的搜索条件
 * 
 * @param searchUuid - 搜索条件 UUID（业务ID）
 * @returns 搜索条件
 */
export async function getSavedSearch(searchUuid: string): Promise<SavedSearch> {
  const response = await apiRequest<SavedSearch>(
    `/saved-searches/${searchUuid}`,
    {
      method: 'GET',
    }
  );
  return response;
}

/**
 * 更新保存的搜索条件
 * 
 * @param searchUuid - 搜索条件 UUID（业务ID）
 * @param data - 更新参数
 * @returns 更新后的搜索条件
 */
export async function updateSavedSearch(
  searchUuid: string,
  data: UpdateSavedSearchParams
): Promise<SavedSearch> {
  const response = await apiRequest<SavedSearch>(
    `/saved-searches/${searchUuid}`,
    {
      method: 'PUT',
      data,
    }
  );
  return response;
}

/**
 * 删除保存的搜索条件
 * 
 * @param searchUuid - 搜索条件 UUID（业务ID）
 */
export async function deleteSavedSearch(searchUuid: string): Promise<void> {
  await apiRequest(
    `/saved-searches/${searchUuid}`,
    {
      method: 'DELETE',
    }
  );
}

File: riveredge-backend/src/infra/schemas/saved_search.py
"""
保存搜索条件 Schema 模块

定义保存搜索条件相关的 Pydantic Schema，用于 API 请求和响应的数据验证
"""

from datetime import datetime
from typing import Optional, Dict, Any, List

from pydantic import BaseModel, Field, ConfigDict


class SavedSearchBase(BaseModel):
    """
    保存搜索条件基础 Schema
    
    包含保存搜索条件的通用字段定义
    
    Attributes:
        page_path: 页面路径
        name: 搜索条件名称
        is_shared: 是否共享给其他用户
        is_pinned: 是否置顶
        search_params: 搜索参数（JSON 存储）
    """
    
    page_path: str = Field(..., max_length=255, description="页面路径")
    name: str = Field(..., min_length=1, max_length=100, description="搜索条件名称")
    is_shared: bool = Field(default=False, description="是否共享给其他用户")
    is_pinned: bool = Field(default=False, description="是否置顶")
    search_params: Dict[str, Any] = Field(default_factory=dict, description="搜索参数（JSON 存储）")


class SavedSearchCreate(SavedSearchBase):
    """
    保存搜索条件创建 Schema
    
    用于创建新保存搜索条件时的数据验证
    """
    pass


class SavedSearchUpdate(BaseModel):
    """
    保存搜索条件更新 Schema
    
    用于更新保存搜索条件信息时的数据验证。
    所有字段都是可选的，只更新提供的字段。
    """
    
    name: Optional[str] = Field(None, min_length=1, max_length=100, description="搜索条件名称")
    is_shared: Optional[bool] = Field(None, description="是否共享给其他用户")
    is_pinned: Optional[bool] = Field(None, description="是否置顶")
    search_params: Optional[Dict[str, Any]] = Field(None, description="搜索参数（JSON 存储）")


class SavedSearchResponse(SavedSearchBase):
    """
    保存搜索条件响应 Schema
    
    用于 API 响应时的数据序列化
    
    Attributes:
        id: 搜索条件 ID
        uuid: 业务ID（UUID，对外暴露）
        tenant_id: 组织 ID
        user_id: 用户 ID
        created_at: 创建时间
        updated_at: 更新时间
    """
    
    id: int = Field(..., description="搜索条件 ID")
    uuid: str = Field(..., description="业务ID（UUID，对外暴露）")
    tenant_id: Optional[int] = Field(None, description="组织 ID")
    user_id: int = Field(..., description="用户 ID")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")
    
    model_config = ConfigDict(from_attributes=True)


class SavedSearchListResponse(BaseModel):
    """
    保存搜索条件列表响应 Schema
    
    用于分页列表查询的响应
    
    Attributes:
        items: 搜索条件列表
        total: 总数量
    """
    
    items: List[SavedSearchResponse] = Field(default_factory=list, description="搜索条件列表")
    total: int = Field(default=0, description="总数量")

